### DISCUSSION TITLE
Floating window statusline override my buffer's statusline

### DISCUSSION THREAD

**Discussion by winkee01** [nvim-lualine/lualine.nvim#882] [Configuration]
When I type `:LspInfo`, it will pop up a floating window, and the status line for this floating window will also show, it overrides my buffer's statusline, how I can disable it and only show my buffer's statusline? because to me, it makes no sense to show a floating windows statusline.

![image](https://user-images.githubusercontent.com/49930785/198801931-067cb7f8-f846-4d00-8798-70e33aad80aa.png)

**Comment by fitrh** [comment]
That's how it works, since `'statusline'` is a local window optons, all
you can do is ignore certains file types by adding them to `options.disable_filetypes`

```lua
require("lualine").setup({
    options = {
        disable_filetypes = { "lspinfo" },
    }
}
```

Or create an extension and compose what you want your status line look like

```lua
local lspinfo = {
    filetypes = { "lspinfo" },
    sections = {
        lualine_c = { "buffers" },
    },
}

require("lualune").setup({
    extensions = { lspinfo },
})
```

Or make sure you don't use global statusline and add the file types
into `options.ignore_focus`

```lua
vim.opt.laststatus = 2 -- you can ignore this as long as options.globalstatus = false (the default)
require("lualine").setup({
    options = {
        ignore_focus = { "lspinfo" },
    }
}
```

**Reply by winkee01** [reply]
that's exactly what I did as shown in the below picture, but by doing that, it still hides my buffer's statusline.

**Reply by fitrh** [reply]
Which one? adding it to `disable_filetypes` or creating an extension?

**Reply by winkee01** [reply]
I tried the first one. which is 
```lua
require("lualine").setup({
    options = {
        disable_filetypes = { "lspinfo" },
    }
}
```

**Reply by fitrh** [reply]
If you create an extension, the just copy your `sections` to your extension `sections`

**Reply by fitrh** [reply]
> I tried the first one. which is
> 
> ```lua
> require("lualine").setup({
>     options = {
>         disable_filetypes = { "lspinfo" },
>     }
> }
> ```

Yes, that disable the statusline for certain file types

**Reply by winkee01** [reply]
I mean, disabling it does not solve the problem, because the statusline area still belongs to it, it is just not showing, but my buffer's statusline is hidden by it.

**Reply by fitrh** [reply]
Then create an extension and just copy your `sections` into you extension `sections`

**Reply by winkee01** [reply]
I tried also, unfortunately, it's the same.

**Reply by fitrh** [reply]
You mean still hidden? if you creata an extension, then remove the file type
from `options.disable_filetypes`

**Reply by winkee01** [reply]
Here is what it looks like after I apply an extension
<img width="886" alt="CleanShot 2022-10-29 at 12 28 47@2x" src="https://user-images.githubusercontent.com/49930785/198813869-6e48ead7-e188-4519-ad00-a2f0dab518e8.png">

**Reply by fitrh** [reply]
Yes, that's how it looks like if you copy the code snippet I provided,
if you want it to be exactly the same as your regular statusline, just
copy your regular `sections` into your extension `sections`

**Reply by winkee01** [reply]
I tried, it does not work at all, the statusline still belongs to the float window, not my buffer.

**Reply by fitrh** [reply]
It seems what you wants is if you are in `lspinfo` buffer, you
want the statusline to show the statusline from the previous buffer?
If that so, you should not using global status line and then
add `lspinfo` to `options.ingnore_focus`

```lua
vim.opt.laststatus = 2
require("lualine").setup({
    options = {
        ignore_focus = { "lspinfo" },
    }
}
```

**Reply by winkee01** [reply]
thank! now it works, but why is that? why does the global status line cause the problem?

**Reply by fitrh** [reply]
IDK, but It seems that `options.ignore_focus` is not handling the global
satatus line poperly yet

**Reply by shadmansaleh** [reply]
@fitrh  I think it should handle global statusline already. can you tell me when it misbehaves?

**Reply by fitrh** [reply]
@shadmansaleh When we run `:LspInfo`, the global statusline is hidden

**Reply by fitrh** [reply]
It seems it only happens when the window is a floating window (?)
I tried adding `help` into `ignore_focus` and yes, it works with
the global status line.

**Reply by shadmansaleh** [reply]
Thanks. I'll check it out.
